### ISSUE TITLE
overprinting @!ab escape not working in gmt text paragraph mode

### ISSUE THREAD

**Issue by tomeracles** [GenericMappingTools/gmt#4222]
Thanks very much for sorting out the problem I identified with the @! escape not being happy with font changes. 😃 But I'm afraid I've run into another problem!

**Description of the problem**

When using gmt text -M (paragraph mode) the @!ab escape sequence does not work the same way it does outside of paragraph mode. I noticed this at first when trying to use the combined epsilon-dot character which can now be made with gmt 6.1.1 using the sequence ```@!\227@~\145@~```.

```
gmt text -JX5c/5c -R0/5/0/5 -B -pdf test1 << EOF
1 1 @!\227@~\145@~
EOF
```

![image](https://user-images.githubusercontent.com/49723355/93336783-231d8300-f820-11ea-84dd-1f0fa4df1bf8.png)

Unfortunately this doesn't seem to work so well in paragraph mode:

```
gmt text -JX5c/5c -R0/5/0/5 -B -M -pdf test2 << EOF
> 1 1 12p 1c c
@!\227@~\145@~
EOF
```

![image](https://user-images.githubusercontent.com/49723355/93336820-2dd81800-f820-11ea-9b19-35cf06df0573.png)

I had another look and noticed something slightly different happens even if you're not dealing with weird characters:

```
gmt text -JX5c/5c -R0/5/0/5 -B -pdf test3 << EOF
1 1 @!at
EOF
```

![image](https://user-images.githubusercontent.com/49723355/93336662-f8332f00-f81f-11ea-831e-a1adef9dde44.png)

```
gmt text -JX5c/5c -R0/5/0/5 -B -M -pdf test4 << EOF
> 1 1 12p 1c c
@!at
EOF
```

![image](https://user-images.githubusercontent.com/49723355/93336701-084b0e80-f820-11ea-87b1-34ef2201c509.png)

Not sure if I have missed something about how to use paragraph mode but this looks like a bug to me.

Thanks
Tom

**System information**

* Operating system:  macOS Mojave 10.14.1
* GMT version: 6.1.1

**Comment by PaulWessel** [comment]
I am not sure I will be able to address this problem.  I wrote most of that PostScript programming back 20 years a go and not it suffers from some lack of comments.  It is pretty impenetrable stuff.  I will try to add more comments to the statements I do understand and hopefully this may help me, or others.
Hi @seisman , I guess it is still best for me to edit the original files like _PSL_text.ps_ and just run _gmt_make_PSL_strings.sh_ to pack them? I want to add more comments mostly, possibly some minor indent changes.

**Comment by seisman** [comment]
> I guess it is still best for me to edit the original files like _PSL_text.ps_ and just run _gmt_make_PSL_strings.sh_ to pack them? I want to add more comments mostly, possibly some minor indent changes.

Yes.

**Comment by PaulWessel** [comment]
So the conclusion to this issue is that it was possible to handle the font change for composite characters in the plain text typesetting, but for the paragraph mode (**-M**) we descend into the madness of reverse polish notation _PostScript_ programmed by me about 20 years ago or more.  I spent some time trying to see the logic but it was a bit futile.  The best I can do was to update the documentation for **pstext -M** to state it does not support composite characters.  Sorry, we tried.
